lib/chef_git/branched_data_bag.rb

```diff
@@ -0,0 +1,39 @@
+require 'chef/node'
+require 'mixlib/shellout'
+
+module ChefGit::BranchedDataBag
+  extend self
+
+  def bag_name(data_bag)
+    if chef_git_environment == 'master'
+      return data_bag
+    elsif data_bag_changed?(data_bag)
+      return "#{chef_git_environment}__#{data_bag}"
+    else
+      return data_bag
+    end
+  end
+
+  private
+
+  def data_bag_changed?(data_bag)
+    Dir.chdir(ChefGit::REPO_PATH) do
+      git_diff = Mixlib::ShellOut.new("git diff origin/master data_bags/#{data_bag}")
+      git_diff.run_command
+      unchanged = !git_diff.error? && git_diff.stdout.empty? && git_diff.stderr.empty?
+      return !unchanged
+    end
+  end
+
+  def chef_git_environment
+    @branch ||= begin
+      # If the branch is specified in client.rb or -E flag, use that
+      if Chef::Config.environment
```

Comment:
seems like that would make the line too long, I'd prefer the verbosity as it's a bit easier to read